proposal/cram.md

```diff
@@ -0,0 +1,95 @@
+# Proposal: CRAM (v3) format support in GO
+
+Last updated: March 23, 2017
+
+## Abstract
+
+[CRAM format](https://samtools.github.io/hts-specs/CRAMv3.pdf) encodes genomic alignments to a reference.
+It has better lossless (and optional lossy) compression compared to existing BAM format. It is also
+becoming more widely used. We propose the addition of a CRAM Reader to [biogo/hts](https://github.com/biogo/hts)
+under `hts/cram`.
+
+## Background
+
+As more projects move to whole-genome sequencing across large cohorts, drive space to store alignments becomes a
+concern. This is the reason for CRAM which has a much smaller footprint than BAM. This is done largely through
+encoding difference to a reference rather than saving the full sequence and by integer compression schemes.
+
+The functionality to be implemented will be driven by the specification: https://samtools.github.io/hts-specs/CRAMv3.pdf
+but limited to the types observed in the wild and implemented and used in [htslib](https://samtools.github.io)
+
+## High-Level API
+
+The `hts/cram` Reader should match the `hts/bam` API as closely as is reasonable.
+
+```Go
+
+import "biogo/hts/cram"
+
+var cr *cram.CRAM 
+var err error
+
+cr, err = cram.NewReader(rdr, io.Reader, rd int, reference *sam.Reference)
+cr.Omit(bam.AllVariableLengthData)
```

Comment:
If we want finer control, this might not want to be a bam package constant, but rather in cram.